src/main/java/io/vertx/proton/sasl/impl/ProtonSaslExternalImpl.java

```diff
@@ -0,0 +1,31 @@
+package io.vertx.proton.sasl.impl;
+
+public class ProtonSaslExternalImpl extends ProtonSaslMechanismImpl {
```

Comment:
The superclass has state so making it a singleton seems a bit off, I think I'd prefer it the way it is.